types/map.go

```diff
@@ -228,3 +230,33 @@ func (m Map) IsNull() bool {
 func (m Map) IsUnknown() bool {
 	return m.Unknown
 }
+
+func (m Map) String() string {
+	if m.Unknown {
+		return attr.UnknownString
+	}
+
+	if m.Null {
+		return attr.NullString
+	}
+
+	// We want the output to be consistent, so we sort the output by key
+	keys := make([]string, 0, len(m.Elems))
+	for k := range m.Elems {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	var res strings.Builder
+
+	res.WriteString("[")
+	for i, k := range keys {
+		if i != 0 {
+			res.WriteString(",")
+		}
+		res.WriteString(fmt.Sprintf("%s:%s", k, m.Elems[k].String()))
```

Comment:
I think we should quote (e.g. `%q`) the keys for clarity, as they are all strings and can contain spaces, braces, commas, colons, etc.

Reply:
👍